### PR TITLE
[Bugfix] Restore Wan2.2 RoPE cache hits

### PR DESCRIPTION
**Pull request by yuweih205**
## Purpose

Wan2.2 and Wan2.2 VACE rebuilt identical rotary embeddings on every denoising step. The cache-miss path wrote the unused `_hidden_states_shape` attribute, while the lookup read `_cached_rope_resolution`.

Both transformers now cache by post-patch resolution, device, and dtype. Repeated forwards reuse the cached tensors; resolution or dtype changes rebuild them. RoPE values and attention layout are unchanged.

The CPU regression test exercises each transformer's real forward with a tiny zero-attention-block model. Four cases cover Wan/VACE × resolution/dtype: two identical forwards call `self.rope` once, changing the key rebuilds the cache, and repeating the new key reuses it. It also checks cache-object identity, tensor shape/dtype, and exact repeated outputs. The checkpoint-only convolution parameters are explicitly initialized. The existing `core_model and cpu` CI lane collects all four cases.

A similar correction appears in draft #6098 as part of its broader PipeFusion work; this PR isolates the cache fix.

## Test Plan

Review follow-up candidate: `ed54da6d`, on top of `a921e7b9` (base `4c7a98c2`).
CPU environment: Python 3.12.3, vLLM 0.29.0, torch 2.13.0+cu130, diffusers 0.40.0, transformers 5.14.1. CUDA was unavailable; no model downloads or GPU are needed for this test.

With the repository's test dependencies installed:

```bash
export CUDA_VISIBLE_DEVICES= VLLM_TARGET_DEVICE=cpu OMP_NUM_THREADS=1 MKL_NUM_THREADS=1
python -m pytest -q tests/diffusion/models/wan2_2/test_wan22_rope_cache.py
python -m pytest -q tests/diffusion/models/wan2_2 -m 'core_model and cpu'
pre-commit run --files tests/diffusion/models/wan2_2/test_wan22_rope_cache.py
```

## Test Result

- New CPU regression module: **4 passed**.
- Complete Wan2.2 CPU directory suite: **179 passed**.
- Negative control: replacing only the two transformer production files with their original versions from `4c7a98c2` makes **all 4 new cases fail** at `rope.call_count == 1` (actual: 2). The same final test fixture is used in both arms.
- With `torch.use_deterministic_algorithms(True)` and `torch.utils.deterministic.fill_uninitialized_memory = True`: **4 passed**, checking that the fixture does not depend on uninitialized allocation contents.
- All applicable changed-file pre-commit hooks passed, including Ruff, Python 3.10 mypy, CI marks, SPDX, and forbidden-import checks.

## Earlier H200 evidence

These measurements were recorded on vLLM 0.28.0 with parent `af74a5a1551e3b809f7d5cddfbba070deb4bb298` and candidate `26992a4f48bf693583fe893b64eb1d896f6b3ba3`; they were not rerun for this test-only follow-up.

Both transformers reused cached objects for an unchanged key, rebuilt them after dtype and resolution changes, returned FP32 RoPE after switching to FP32, and produced exact repeated outputs.

For the targeted zero-block forward, 8 same-resolution steps, median of 10 interleaved H200 samples: **6.644 ms** with a forced miss each step versus **5.891 ms** with the cache (**−11.32%**, **8 → 1** RoPE calls). This measures the transformer front-end path, not full-model end-to-end latency.

AI assistance for this review follow-up: Codex added the regression test, ran the positive/negative controls and CPU/pre-commit checks above, and updated this validation record.

### CODE REVIEW

**Comment by yuweih205** [comment]
Self-review for `26992a4f`: I checked that both the Wan2.2 and VACE transformers now key the RoPE cache on post-patch resolution, device, and dtype, and that the cache-miss path writes that key instead of the unused `_hidden_states_shape` attribute, so repeated steps with the same post-patch resolution, device, and dtype reuse the cached tuple. The `self.rope` call, the even/odd slicing, and the dtype cast on the miss path are unchanged, and the hit path still returns the stored tuple, so RoPE values and attention layout are unchanged.

Ruff, the changed-file pre-commit checks, and the Wan2.2 CPU unit suite (75 passed) are green. The H200 checks confirmed reuse of the same cached objects for an unchanged key, rebuilds after resolution and dtype changes, FP32 RoPE after switching to FP32, and exact repeated outputs in both transformers. The `-11.32%` (`8 -> 1` RoPE calls) figure is a targeted zero-block front-end measurement over 8 same-resolution steps and does not represent full-model end-to-end latency.

**Comment by yuweih205** [comment]
Gentle ping — this has been open 11 days with CI green and a self-review posted, and no reviewer assigned yet. Per the contributing guide I am pinging rather than letting it sit.

It is a small, self-contained change to Wan2.2's RoPE cache; the description carries the measurements and the test evidence. Could a maintainer take a look, or add the `ready` label so the full pipeline runs? Happy to rebase or split it if that helps.

**Comment by Dong1017** [comment]
Verified on H200 against both trees (head `26992a4f`, base `af74a5a`): repeated same-key forwards now issue 1 `self.rope` call and the cached tuple is bitwise-equal to a fresh recompute in both transformers; resolution and fp32-cast changes miss and then re-hit; base reproduces the never-hit defect (0% benefit). Front-end A/B: 9.80% faster with the cache vs forced-miss each step, matching the reported 11.32%. No correctness concerns with the (resolution, device, dtype) key — it covers every input `WanRotaryPosEmbed` output depends on, and all cos/sin consumers (FA/HIP/NPU/native, SP sharding) are read-only, so reusing the same tensors across steps is safe.

Two asks: (1) add the call-count regression test from your own reproduction recipe (two identical forwards → 1 rope call; plus dtype/resolution invalidation) to the existing CPU lane — this defect class is exactly "renamed attribute silently kills the cache," and it ran unnoticed for a long time; (2) #6098 carries the same fix inside its broader scope — this PR should land first and #6098 drop its copy on rebase.

FYI the in-tree S2V transformer keys its cache on `(batch, seq, device)` — fine there because its rope depends on exactly those; not a defect, noting for anyone comparing the three sites later.

File: vllm_omni/diffusion/models/wan2_2/wan2_2_transformer.py [hunk 1]
```diff
@@ -1081,12 +1081,13 @@ def forward(
 
         # Compute RoPE embeddings (sharded by _sp_plan via split_output=True)
         current_rope_resolution = (post_patch_num_frames, post_patch_height, post_patch_width)
-        if self._cached_rope_resolution == current_rope_resolution and self._cached_rope_emb is not None:
+        cache_key = (*current_rope_resolution, hidden_states.device, hidden_states.dtype)
```
Comment (Dong1017) [inline vllm_omni/diffusion/models/wan2_2/wan2_2_transformer.py:1084]: Could you add a regression test for the never-hit cache? The recipe is already in your PR body: same-key forward x2 -> assert `self.rope` called once, then resolution/dtype change -> assert rebuild. CPU-only works here (wrap the model in `set_current_vllm_config` + a `_PP` stub like the existing wan2_2 CPU tests). Without it, the next attribute rename re-breaks this invisibly.

(Minor, non-blocking: this block now exists in three files — main/VACE/S2V — and VACE inherits the main class, so a shared base-class helper would remove two copies. Fine as a follow-up.)
